stdlib/public/core/MutableCollection.swift

```diff
@@ -529,24 +529,24 @@ extension MutableCollection {
 @inlinable
 @_preInverseGenerics
 public func swap<T: ~Copyable>(_ a: inout T, _ b: inout T) {
-  // Semantically equivalent to (a, b) = (b, a).
-  // Microoptimized to avoid retain/release traffic.
-#if $BuiltinUnprotectedAddressOf
-  let p1 = Builtin.unprotectedAddressOf(&a)
-  let p2 = Builtin.unprotectedAddressOf(&b)
-#else
-  let p1 = Builtin.addressof(&a)
-  let p2 = Builtin.addressof(&b)
-#endif
-  _debugPrecondition(
-    p1 != p2,
-    "swapping a location with itself is not supported")
```

Comment:
I love the new formulation (it's finally here!), but we are losing this explanatory assertion. We could have
```
_debugPrecondition(
  Builtin.unprotectedAddressOf(&a) != Builtin.unprotectedAddressOf(&b),
```

Reply:
That's guaranteed in a much better way by Swift's exclusivity rules. I don't think we'd need/want to double check the compiler's work here.

Reply:
Good point! I guess this must have only been a hangover from Swift 3, then.

Reply:
Yep! This code originated in 2015, in the Swift 2 era. 🦕